diff --git a/node_modules/expo-dev-launcher/ios/ReactDelegateHandler/ExpoDevLauncherReactDelegateHandler.swift b/node_modules/expo-dev-launcher/ios/ReactDelegateHandler/ExpoDevLauncherReactDelegateHandler.swift
index e02ea0c..e2e4c97 100644
--- a/node_modules/expo-dev-launcher/ios/ReactDelegateHandler/ExpoDevLauncherReactDelegateHandler.swift
+++ b/node_modules/expo-dev-launcher/ios/ReactDelegateHandler/ExpoDevLauncherReactDelegateHandler.swift
@@ -6,10 +6,44 @@ import React
 
 private class DevLauncherWrapperView: UIView {
   weak var devLauncherViewController: UIViewController?
+  private var devLauncherConstraints: [NSLayoutConstraint] = []
+
+  @objc
+  func frameDidChange() {
+    if let controller = devLauncherViewController {
+      setDevLauncherViewControllerConstraints(controller)
+    }
+  }
+
+  func setDevLauncherViewControllerConstraints(_ viewController: UIViewController) {
+    viewController.view.translatesAutoresizingMaskIntoConstraints = false
+    if !devLauncherConstraints.isEmpty {
+      NSLayoutConstraint.deactivate(devLauncherConstraints)
+    }
+    devLauncherConstraints = [
+      viewController.view.topAnchor.constraint(equalTo: self.topAnchor),
+      viewController.view.leadingAnchor.constraint(equalTo: self.leadingAnchor),
+      viewController.view.trailingAnchor.constraint(equalTo: self.trailingAnchor),
+      viewController.view.bottomAnchor.constraint(equalTo: self.bottomAnchor)
+    ]
+    NSLayoutConstraint.activate(devLauncherConstraints)
+  }
 
 #if !os(macOS)
+  override var frame: CGRect {
+    didSet {
+      UIView.animate(withDuration: 1.0, animations: {
+        self.frameDidChange()
+      })
+    }
+  }
   override func didMoveToWindow() {
     super.didMoveToWindow()
+    #if os(iOS)
+    if window != nil {
+      NotificationCenter.default.addObserver(self, selector: #selector(frameDidChange), name: UIDevice.orientationDidChangeNotification, object: nil)
+    }
+    #endif
 
     guard let devLauncherViewController,
       let window,
@@ -33,6 +67,9 @@ private class DevLauncherWrapperView: UIView {
     if newWindow == nil {
       devLauncherViewController?.willMove(toParent: nil)
       devLauncherViewController?.removeFromParent()
+      #if os(iOS)
+      NotificationCenter.default.removeObserver(self, name: UIDevice.orientationDidChangeNotification, object: nil)
+      #endif
     }
   }
 #endif
@@ -78,14 +115,7 @@ public class ExpoDevLauncherReactDelegateHandler: ExpoReactDelegateHandler, EXDe
     let wrapperView = DevLauncherWrapperView()
     wrapperView.devLauncherViewController = viewController
     wrapperView.addSubview(viewController.view)
-    viewController.view.translatesAutoresizingMaskIntoConstraints = false
-    NSLayoutConstraint.activate([
-      viewController.view.topAnchor.constraint(equalTo: wrapperView.topAnchor),
-      viewController.view.leadingAnchor.constraint(equalTo: wrapperView.leadingAnchor),
-      viewController.view.trailingAnchor.constraint(equalTo: wrapperView.trailingAnchor),
-      viewController.view.bottomAnchor.constraint(equalTo: wrapperView.bottomAnchor)
-    ])
-
+    wrapperView.setDevLauncherViewControllerConstraints(viewController)
     return wrapperView
   }
 
